src/mono/wasm/build/WasmApp.Native.targets

```diff
@@ -628,7 +628,10 @@
       <_WasmAssembliesInternal Remove="@(_WasmAssembliesInternal)" />
 
       <_WasmAOTSearchPaths Include="$(MicrosoftNetCoreAppRuntimePackRidNativeDir)" />
-      <_WasmAOTSearchPaths Include="$(MicrosoftNetCoreAppRuntimePackRidDir)\lib\net8.0" />
+
+      <!-- find the path with the assemblies, so we don't have to hardcode the tfm -->
+      <_SystemIOPath Include="$(MicrosoftNetCoreAppRuntimePackRidNativeDir)\lib\*\System.IO.dll" />
```

Comment:
This doesn't look right. You want to save the path in a property and use the RuntimePackAsset item instead of the glob:

```
<PropertyGroup>
  <_SystemPrivateCoreLibPath>@(RuntimePackAsset->WithMetadataValue('Filename', 'System.Private.CoreLib'))</_SystemPrivateCoreLibPath>
</PropertyGroup>

<ItemGroup>
  <_WasmAOTSearchPaths Include="$([System.IO.Path]::GetDirectoryName('$(_SystemPrivateCoreLibPath)'))" />
</ItemGroup>
```

Reply:
> This doesn't look right. You want to save the path in a property, not an item.

I was using it for the glob.

> Also, use the RuntimePackAsset item instead of the glob:

Much better idea, thanks!